core/trino-main/src/test/java/io/trino/execution/scheduler/TestScaledWriterScheduler.java

```diff
@@ -0,0 +1,490 @@
+        public TaskLifecycleListener getTaskLifecycleListener()
+        {
+            throw new UnsupportedOperationException();
+        }
+
+        @Override
+        public void schedulingComplete()
+        {
+            throw new UnsupportedOperationException();
+        }
+
+        @Override
+        public void schedulingComplete(PlanNodeId partitionedSource)
+        {
+            throw new UnsupportedOperationException();
+        }
+
+        @Override
+        public void cancel()
+        {
+            throw new UnsupportedOperationException();
+        }
+
+        @Override
+        public void abort()
+        {
+            throw new UnsupportedOperationException();
+        }
+
+        @Override
+        public void recordGetSplitTime(long start)
+        {
+            throw new UnsupportedOperationException();
+        }
+
+        @Override
+        public Optional<RemoteTask> scheduleTask(InternalNode node, int partition, Multimap<PlanNodeId, Split> initialSplits)
+        {
+            return Optional.of(new TestingRemoteTask(TaskId.valueOf("taskId"), "nodeId", fragment));
+        }
+
+        @Override
+        public void failTask(TaskId taskId, Throwable failureCause)
+        {
+            throw new UnsupportedOperationException();
+        }
+
+        @Override
+        public List<RemoteTask> getAllTasks()
+        {
+            throw new UnsupportedOperationException();
+        }
+
+        @Override
+        public List<TaskStatus> getTaskStatuses()
+        {
+            throw new UnsupportedOperationException();
+        }
+
+        @Override
+        public Optional<ExecutionFailureInfo> getFailureCause()
+        {
+            throw new UnsupportedOperationException();
+        }
+    }
+
+    private static class TestingNodeSelector
+            implements NodeSelector
+    {
+        private NodeSelector nodeSelector;
+
+        public TestingNodeSelector(NodeSelector nodeSelector)
+        {
+            this.nodeSelector = nodeSelector;
+        }
+
+        @Override
+        public void lockDownNodes() {}
+
+        @Override
+        public List<InternalNode> allNodes()
+        {
+            return null;
+        }
+
+        @Override
+        public InternalNode selectCurrentNode()
+        {
+            return null;
+        }
+
+        @Override
+        public List<InternalNode> selectRandomNodes(int limit, Set<InternalNode> excludedNodes)
+        {
+            return nodeSelector.selectRandomNodes(limit, ImmutableSet.of());
+        }
+
+        @Override
+        public SplitPlacementResult computeAssignments(Set<Split> splits, List<RemoteTask> existingTasks)
+        {
+            return null;
+        }
+
+        @Override
+        public SplitPlacementResult computeAssignments(Set<Split> splits, List<RemoteTask> existingTasks, BucketNodeMap bucketNodeMap)
+        {
+            return null;
+        }
+    }
+
+    private static PlanFragment createFragment()
```

Comment:
I don't think this is required for our use case right?

Reply:
why? it is needed to call scheduleTask. it would be not needed if we directly test getNewTaskCount which was my initial approach.

Reply:
that makes sense